src/detectors.rs

```diff
@@ -0,0 +1,329 @@
+//! Utility to help with "really nice to add a warning for tasks that might be blocking"
+use libc;
+use rand::thread_rng;
+use rand::Rng;
+use std::collections::HashSet;
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex};
+use std::thread::ThreadId;
+use std::time::Duration;
+use std::{env, thread};
+use tokio::runtime::{Builder, Runtime};
+
+const PANIC_WORKER_BLOCK_DURATION_DEFAULT: Duration = Duration::from_secs(60);
+
+fn get_panic_worker_block_duration() -> Duration {
+    let duration_str = env::var("MY_DURATION_ENV").unwrap_or_else(|_| "60".to_string());
+    duration_str
+        .parse::<u64>()
+        .map(Duration::from_secs)
+        .unwrap_or(PANIC_WORKER_BLOCK_DURATION_DEFAULT)
+}
+
+#[cfg(unix)]
+fn get_thread_id() -> libc::pthread_t {
+    unsafe { libc::pthread_self() }
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct ThreadInfo {
+    pub id: ThreadId,
+    #[cfg(unix)]
+    pub pthread_id: libc::pthread_t,
```

Comment:
i think that, rather than making these fields `pub`, we should probably make them private and add public accessor methods. this way, users don't try to match on a type that has `#[cfg(unix)]` fields, which could result in compilation errors on other platforms. hiding these fields behind methods would make it a little harder to accidentally write code that only compiles on unix.